src/main/java/de/retest/recheck/cli/subcommands/Commit.java

```diff
@@ -102,6 +128,39 @@ private void applyChanges( final Persistence<SutState> persistence, final SuiteC
 		}
 	}
 
+	private void checkForWarningMessage( final ReviewResult changeSets ) {
+		for ( final SuiteChangeSet suiteChangeSet : changeSets.getSuiteChangeSets() ) {
+			for ( final TestChangeSet testChangeSet : suiteChangeSet.getTestChangeSets() ) {
+				final ActionChangeSet actionChangeSet = testChangeSet.getInitialStateChangeSet();
+				final AttributeChanges attributeChanges = actionChangeSet.getIdentAttributeChanges();
+				for ( final Set<AttributeDifference> attributeDifferences : attributeChanges.getChanges().values() ) {
+					for ( final AttributeDifference attributeDifference : attributeDifferences ) {
+						final ElementIdentificationWarning warning =
+								attributeDifference.getElementIdentificationWarning();
+						if ( warning != null ) {
```

Comment:
Rather than doing this nested `for` and `if` stuff, maybe you want to use `flatMap` and `filter`?